src/main/java/lotto/domain/Lotto.java

```diff
@@ -0,0 +1,76 @@
+package lotto.domain;
+
+import static lotto.constants.ErrorMessage.ERROR_LOTTO_NUMBER_DUPLICATE;
+import static lotto.constants.ErrorMessage.ERROR_LOTTO_NUMBER_RANGE_OVER;
+import static lotto.constants.ErrorMessage.ERROR_LOTTO_NUMBER_SIZE;
+import static lotto.constants.LottoConfiguration.LOTTO_SIZE;
+import static lotto.constants.LottoConfiguration.MAX_RANGE;
+import static lotto.constants.LottoConfiguration.MIN_RANGE;
+
+import java.util.Collections;
+import java.util.List;
+import lotto.exception.LottoException;
+
+public class Lotto {
+    private final List<Integer> numbers;
+
+    public Lotto(List<Integer> numbers) {
+        validate(numbers);
+        this.numbers = numbers;
+    }
+
+    private void validate(List<Integer> numbers) {
+        NumberSizeCheck(numbers);
+        DuplicateNumberCheck(numbers);
+        NumberRangeCheck(numbers);
+    }
+
+    private void NumberSizeCheck(List<Integer> numbers) {
+        if (numbers.size() != LOTTO_SIZE.getConfig()) {
+            throw LottoException.of(ERROR_LOTTO_NUMBER_SIZE);
+        }
+    }
+
+    private void DuplicateNumberCheck(List<Integer> numbers) {
+        long count = numbers
+                .stream()
+                .distinct()
+                .count();
+
+        if (numbers.size() != count) {
+            throw LottoException.of(ERROR_LOTTO_NUMBER_DUPLICATE);
+        }
+    }
+
+    private void NumberRangeCheck(List<Integer> numbers) {
+        boolean overRangeResult = numbers
+                .stream()
+                .anyMatch(num -> num < MIN_RANGE.getConfig() || num > MAX_RANGE.getConfig());
+
+        if (overRangeResult) {
+            throw LottoException.of(ERROR_LOTTO_NUMBER_RANGE_OVER);
+        }
+    }
+
+    public boolean isContainsNumber(int number) {
+        return numbers.contains(number);
+    }
+
+    public long lottoNumberComparison(Lotto otherLotto) {
```

Comment:
로또와 다른 로또를 비교해서 같은 숫자의 개수를 반환하는 로직이네요!
현재 우리가 구현하고 있는 로또 게임은 6자리의 숫자를 가지고 있기 때문에 이 메서드가 반환할 수 있는 값은 0~6 이라는 범위겠군요.
이 경우 `short`를 굳이 사용할 필요까진 없겠지만, `long`이라는 자료형은 리턴 값에 비해 너무 큰 것 같아요!
자료형을 캐스팅을 통해 변경해 주면 더 좋을 것 같아요!
```suggestion
    public int lottoNumberComparison(Lotto otherLotto) {
```

Reply:
맞아요. 너무 큰 자료형을 사용하고 있었던거 같습니다.
좋은 지적 감사합니다.
한번더 로직을 구현 하면서 메모리적인 방면도 더 생각해보게 되는 시간이 된거 같습니다.